pkg/apis/workspaces/v1alpha2/commands.go

```diff
@@ -115,9 +115,9 @@ type ExecCommand struct {
 	//
 	// Special variables that can be used:
 	//
-	//  - `$PROJECTS_ROOT`: A path where projects sources are mounted
+	//  - `$PROJECTS_ROOT`
 	//
-	//  - `$PROJECT_SOURCE`: A path to a project source ($PROJECTS_ROOT/<project-name>). If there are multiple projects, this will point to the directory of the first one.
+	//  - `$PROJECT_SOURCE`
```

Comment:
I'm not sure about removing the descriptions of the variables here is a good idea (at least, without some other, centralized location we can link to to explain it). Currently, the only way to figure out what `$PROJECTS_ROOT` and `$PROJECT_SOURCE` are to be used for is by checking the description for `env`, where the main purpose is to tell you that you can't override them. Basically, my thinking works out to:
- `ExecCommand` (the use-site): Describe what the variables can be used for
- `Container` (the define-site): Describe which variables can be overridden (or not)

An alternative if we want to keep the command descriptions terse is to e.g. link to a docs page (once a description of reserved env vars is added -- e.g. [here](https://devfile.github.io/devfile/assembly_making-a-workspace-portable-using-a-devfile.html#_adding_commands_to_a_devfile)).

Alternatively, a more generic doc would remove the "Special variables..." entirely; isn't it the case that you can use *any* environment variable in a command? The only difference for `$PROJECTS_ROOT` and `$PROJECT_SOURCE` is that they're supposed to automatically be set, but you should be able to reference e.g. `$M2_HOME`, etc. here as well.

```
// The actual command-line-string. Note environment variables from the container the command runs 
// in can be used in the command, e.g. 
//
// `cd ${PROJECT_SOURCE} && mvn clean install` 
```

I'm less certain this applies to the `workingDir`, where I think documenting the automatically-defined env vars is useful.

Reply:
That makes sense. I've reverted the changes and just kept the section about reservation in env.

I've opened https://github.com/devfile/api/issues/138 to discuss and track on how we can define special variables in devfile spec context, as that seems to be out of this PR's scope.